### ISSUE TITLE
smartctl (6.6) in ceph/ceph:v15.2.5 too old

### ISSUE THREAD

**Issue by shanehsu** [ceph/ceph-container#1747]
**Bug Report**

What happened: Ceph images (used to deploy Ceph with `cephadm`) contains old version (6.6) of smartctl that does not support JSON output.

What you expected to happen: Images used to support official orchestrator should include versions of `smartctl` that is compatible with Ceph.

How to reproduce it (minimal and precise):

```
docker run -it --rm ceph/ceph:v15.2.5 /bin/bash -c "smartctl -a --json /dev/null"
smartctl 6.6 2017-11-05 r4594 [x86_64-linux-5.4.0-42-generic] (local build)
Copyright (C) 2002-17, Bruce Allen, Christian Franke, www.smartmontools.org

=======> UNRECOGNIZED OPTION: json

Use smartctl -h to get a usage summary
```

**Environment**:
* OS (e.g. from /etc/os-release): Ubuntu 20.04.1 LTS
* Kernel (e.g. `uname -a`): 5.4.0-42-generic #46-Ubuntu SMP Fri Jul 10 00:24:02 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
* Docker version (e.g. `docker version`): 19.03.8
* Ceph version (e.g. `ceph -v`): 15.2.5

**Comment by tchaikov** [comment]
@shanehsu does aa644a5fd03aecb9eda8b95002d25ded9c57b25e address this issue?

**Comment by shanehsu** [comment]
@tchaikov  Probably, was going to add some information about versions not being available in official repos. Didn't look into PRs sorry.

Since the fix is committed, should we close it or wait until an image is published? 

Also, will an image be published with the fix with v15.2.5 or do I have to build one by myself?

**Comment by dsavineau** [comment]
> Since the fix is committed, should we close it or wait until an image is published?

We can close it.

> Also, will an image be published with the fix with v15.2.5 or do I have to build one by myself?

Current images (like v15.2.5) won't be rebuild for this.

There's only 2 way to get a newer build.

 - a new base container image (like the next CentOS 8.3)
 - a new ceph release (like the next 15.2.6)

So the change be part of 15.2.6 by default.

Until this, you can either use:
  - `docker.io/ceph/daemon-base:latest-octopus` but the tag could change very often
  - `docker.io/ceph/daemon-base:v5.0.4-stable-5.0-octopus-centos-8` which is a static tag based on the v5.0.4 ceph-container release I've made yesterday and that includes the smartmontools fix [1]
  - rebuild 15.2.5 on your own with the fix.

[1] https://github.com/ceph/ceph-container/releases/tag/v5.0.4
